presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -164,44 +160,8 @@ else if (leftNDV <= rightNDV) {
             }
         }
 
-        return outputStats;
-    }
-
-    @VisibleForTesting
-    PlanNodeStatsEstimate addAntiJoinStats(PlanNodeStatsEstimate joinStats, PlanNodeStatsEstimate antiJoinStats, Set<Symbol> joinSymbols)
-    {
-        checkState(joinStats.getSymbolsWithKnownStatistics().containsAll(antiJoinStats.getSymbolsWithKnownStatistics()));
-
-        double joinOutputRowCount = joinStats.getOutputRowCount();
-        double antiJoinOutputRowCount = antiJoinStats.getOutputRowCount();
-        double totalRowCount = joinOutputRowCount + antiJoinOutputRowCount;
-        PlanNodeStatsEstimate outputStats = joinStats.mapOutputRowCount(rowCount -> rowCount + antiJoinOutputRowCount);
-
-        for (Symbol symbol : antiJoinStats.getSymbolsWithKnownStatistics()) {
-            outputStats = outputStats.mapSymbolColumnStatistics(symbol, joinColumnStats -> {
-                SymbolStatsEstimate antiJoinColumnStats = antiJoinStats.getSymbolStatistics(symbol);
-                // weighted average
-                double newNullsFraction = (joinColumnStats.getNullsFraction() * joinOutputRowCount + antiJoinColumnStats.getNullsFraction() * antiJoinOutputRowCount) / totalRowCount;
-                double distinctValues;
-                if (joinSymbols.contains(symbol)) {
```

Comment:
NDVs should only be added for join columns and for uncorrelated columns it should stay the same. Think about it:
We sum joined rows and unjoined rows by join key. Then, for join key column we should account for NDV from both joined and unjoined rows
Other columns (uncorrelated) are equally represented in both joined an unjoined rows. We shouldn't sum their NDVs.